marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -5759,17 +5760,22 @@ private RequestBody makeRequestBody(CallField[] params) {
                 );
           }
         } else if (param instanceof SingleNodeCallField) {
-          AbstractWriteHandle paramValue = ((SingleNodeCallField) param).getParamValue();
+          AbstractWriteHandle paramValue = (AbstractWriteHandle) ((SingleNodeCallField) param).getParamValue();
           if (paramValue != null) {
             HandleImplementation handleBase = HandleAccessor.as(paramValue);
             if(! handleBase.isResendable()) {
-              hasStreamingPartCondition.set();
+                if(param instanceof BufferedSingleNodeCallField) {
+                    ((BufferedSingleNodeCallField) param).setParamValue(new BytesHandle((BufferableHandle) paramValue));
+                    handleBase = HandleAccessor.as(((SingleNodeCallField) param).getParamValue());
```

Comment:
Would this section require fewer casts and thus be easier to understand if a variable captured the BytesHandle?  As in:

     BytesHandle bytesHandle = new BytesHandle((BufferableHandle) handleBase);
     ((BufferedSingleNodeCallField) param).setParamValue(bytesHandle);
    handleBase = bytesHandle;